### DISCUSSION TITLE
Removing UTM tags when stats are logged

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#1853] [Anything Else]
"I’m using UTM tags, and I’ve recently got feedback that people find it annoying to remove the tags before sharing a link with their clients. Would it be possible to remove the UTM tags from the url as soon as the page view has been logged in Plausible? I think location.replace() would be the best option, as to not pollute the browser history.

https://developer.mozilla.org/en-US/docs/Web/API/Location/replace

Let me know if it’s something you could add on your end, or if there’s a way for me to implement this."

(idea via email)

**Comment by ukutaht** [comment]
Something they could try if they know JS:

1. Use `plausible.manual.js` script instead of the automatic one
2. Trigger the pageview manually with `window.plausible('pageview')`
3. Clean the URL manually in the callback: `window.plausible('pageview', {callback: cleanUrlFromUtmParams})`
